CMakeLists.txt

```diff
@@ -0,0 +1,168 @@
+if(WINDOWS_STORE AND CMAKE_SYSTEM_VERSION STREQUAL 10.0)
+    add_definitions(-DREALM_UWP=1 -D_SCL_SECURE_NO_WARNINGS)
+endif()
+
+# Coverage-specific build flags
+option(REALM_COVERAGE "Add coverage support." OFF)
+if(REALM_COVERAGE)
+  if(MSVC)
+    message(FATAL_ERROR
+            "Code coverage is not yet supported on Visual Studio builds")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage -fprofile-arcs -ftest-coverage -fno-inline -fno-elide-constructors")
+  endif()
+endif()
+
+# Platform-specific build configuration
+if(APPLE)
+    find_library(Foundation Foundation)
+
+    # Enable Apple-specific platforms on Xcode
+    if(CMAKE_GENERATOR STREQUAL Xcode)
+        unset(CMAKE_XCODE_ATTRIBUTE_INSTALL_PATH)
+
+        set(CMAKE_XCODE_ATTRIBUTE_SUPPORTED_PLATFORMS
+            "macosx iphoneos iphonesimulator appletvos appletvsimulator watchos watchsimulator"
+        )
+        set(CMAKE_XCODE_EFFECTIVE_PLATFORMS
+            "-macosx;-iphoneos;-iphonesimulator;-appletvos;-appletvsimulator;-watchos;-watchsimulator"
+        )
+        set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY "libc++")
+
+        # Deployment targets
+        set(CMAKE_XCODE_ATTRIBUTE_MACOSX_DEPLOYMENT_TARGET "10.8")
+        set(CMAKE_XCODE_ATTRIBUTE_IPHONEOS_DEPLOYMENT_TARGET "7.0")
+        set(CMAKE_XCODE_ATTRIBUTE_WATCHOS_DEPLOYMENT_TARGET "2.0")
+        set(CMAKE_XCODE_ATTRIBUTE_TVOS_DEPLOYMENT_TARGET "9.0")
+
+        # Architectures. Everything but iOS uses the default architectures.
+        set(CMAKE_XCODE_ATTRIBUTE_ARCHS[sdk=iphoneos*] "\$(ARCHS_iphoneos_\$(CONFIGURATION))")
+        set(CMAKE_XCODE_ATTRIBUTE_ARCHS_iphoneos_Debug "armv7 arm64")
+        set(CMAKE_XCODE_ATTRIBUTE_ARCHS_iphoneos_MinSizeDebug "armv7 arm64")
+        # iOS release configurations need to build for armv7s for sake of CocoaPods.
+        set(CMAKE_XCODE_ATTRIBUTE_ARCHS_iphoneos_Release "armv7 armv7s arm64")
+        set(CMAKE_XCODE_ATTRIBUTE_ARCHS_iphoneos_RelWithDebInfo "armv7 armv7s arm64")
+        set(CMAKE_XCODE_ATTRIBUTE_ARCHS_iphoneos_MinSizeRel "armv7 armv7s arm64")
+
+        # Bitcode. May be overridden by Rakefile.
+        set(CMAKE_XCODE_ATTRIBUTE_ENABLE_BITCODE[sdk=iphone*] "YES")
+        set(CMAKE_XCODE_ATTRIBUTE_ENABLE_BITCODE[sdk=watch*] "YES")
+        set(CMAKE_XCODE_ATTRIBUTE_ENABLE_BITCODE[sdk=appletv*] "YES")
+        set(CMAKE_XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE "\$(BITCODE_GENERATION_MODE_\$(CONFIGURATION))")
+        set(CMAKE_XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE_Debug "marker")
+        set(CMAKE_XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE_MinSizeDebug "marker")
+        set(CMAKE_XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE_Release "bitcode")
+        set(CMAKE_XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE_RelWithDebInfo "bitcode")
+        set(CMAKE_XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE_MinSizeRel "bitcode")
+    endif()
+elseif(ANDROID)
+  set(ANDROID_OPENSSL_VERSION "1.0.1t")
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Os")
```

Comment:
They are automatically included by the toolchain when the GNU STL is used, which is the default one.